### PR TITLE
[lodash] fix `WeakMap` when targeting es2023

### PR DESCRIPTION
**Pull request by DetachHead**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: [see "3. Symbols as WeakMap keys"](https://dev.to/jasmin/what-is-new-in-es2023-4bcm)
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

fixes #66122

### CODE REVIEW

**Comment by DetachHead** [comment]
sorry for the ping everyone else, this is only supposed to contain a lodash change. no idea how these old commits ended up in this branch but they're gone now

File: types/lodash/tsconfig.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
     "compilerOptions": {
         "module": "commonjs",
         "lib": [
-            "es6"
+            "es2023"
```
Comment (aj-r) [inline types/lodash/tsconfig.json:5]: Does this mean lodash users can no longer target ES <2023?

Comment (DetachHead) [reply types/lodash/tsconfig.json:5]: i don't think so, but i have no idea how to get the `WeakMap` type to be compatible on both targets

Comment (thorn0) [reply types/lodash/tsconfig.json:5]: It might be time to remove this declaration. The comment says it's for es5 compatibility.

**Comment by DetachHead** [comment]
happy to add a test for this change but not really sure how. would adding `es2023` back to `compilerOptions.lib` along with `es6` be the way to do it?

**Review by thorn0 — APPROVED** [review]

**Review by rbuckton — APPROVED** [review]

**Comment by DetachHead** [comment]
Ready to merge

**Pull request merged**